src/Features/LanguageServer/Protocol/Handler/Highlights/DocumentHighlightHandler.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.DocumentHighlighting;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler
+{
+    [Shared]
+    [ExportLspMethod(Methods.TextDocumentDocumentHighlightName)]
+    internal class DocumentHighlightsHandler : IRequestHandler<TextDocumentPositionParams, DocumentHighlight[]>
+    {
+        public async Task<DocumentHighlight[]> HandleRequestAsync(Solution solution, TextDocumentPositionParams request,
```

Comment:
>DocumentHighlight[] [](start = 26, length = 19)

Any reason not to use ImmutableArray?

Reply:
LSP package defines the types as array - https://devdiv.visualstudio.com/DevDiv/_git/VSExtensibility?path=%2Fsrc%2Fproduct%2FcommonLanguageServer%2FRemoteLanguage.Protocol%2FMethods.cs&version=GBdevelop&line=285&lineStyle=plain&lineEnd=286&lineStartColumn=1&lineEndColumn=1  I guess we would ask them why not immutable array.

Since liveshare is using the LSP package as well, it depends on them being the same type.

Reply:
That seems unfortunate. Any idea why the package is not using ImmutableArray? @heejaechang

Reply:
@tinaschrepfer might be able to offer some insight